maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/main/java/org/apache/maven/scm/provider/git/gitexe/command/GitCommandLineUtils.java

```diff
@@ -90,28 +95,36 @@ public static void addTarget( Commandline cl, List<File> files )
     }
 
    /**
-    *
+    * Use this only for command not requiring environment variables (i.e. local commands).
     * @param workingDirectory
     * @param command
     * @return TODO
     */
    public static Commandline getBaseGitCommandLine( File workingDirectory, String command )
    {
-       return getBaseGitCommandLine( workingDirectory, command, Collections.emptyMap() );
+       return getBaseGitCommandLine( workingDirectory, command, null, null );
    }
 
     /**
-     *
+     * Use this for commands requiring environment variables (i.e. remote commands).
      * @param workingDirectory
      * @param command
      * @param environment
      * @return TODO
      */
     public static Commandline getBaseGitCommandLine( File workingDirectory, String command,
+                                                     GitScmProviderRepository repository,
                                                      Map<String, String> environment )
     {
         Commandline cl = getAnonymousBaseGitCommandLine( workingDirectory, command );
-        environment.forEach( cl::addEnvironment );
+        if ( repository != null ) 
+        {
+            prepareEnvVariablesForRepository( repository, environment ).forEach( cl::addEnvironment );
+        }
+        else if ( environment != null )
+        {
+            environment.forEach( cl::addEnvironment );
+        }
```

Comment:
I don't understand this double cascade. Everything from `environment` is ready in the map if repo is not null, thus the `forEach` is redundant, no?

Reply:
This is the replacement of former line 114. What other approach do you suggest to populate cl environment?

Reply:
argggggggggggggg...Commandline isn't flexible enough here. Looks OK to me since we cannot get a map from `Commandline`.